File: Application/src/main/java/com/shareyourproxy/api/rx/command/eventcallback/UserContactAddedEventCallback.java
package com.shareyourproxy.api.rx.command.eventcallback;

import android.os.Parcel;

import com.shareyourproxy.api.domain.model.User;

/**
 * Created by Evan on 6/8/15.
 */
public class UserContactAddedEventCallback extends UserEventCallback {
    private final static java.lang.ClassLoader CL =
        UserContactAddedEventCallback.class.getClassLoader();
    public static final Creator<UserContactAddedEventCallback> CREATOR =
        new Creator<UserContactAddedEventCallback>() {
            @Override
            public UserContactAddedEventCallback createFromParcel(Parcel in) {
                return new UserContactAddedEventCallback(
                    (User) in.readValue(CL),(String) in.readValue(CL));
            }

            @Override
            public UserContactAddedEventCallback[] newArray(int size) {
                return new UserContactAddedEventCallback[size];
            }
        };
    public final String contactId;

    public UserContactAddedEventCallback(User user, String contactId) {
        super(user);
        this.contactId = contactId;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(contactId);
    }
}

File: Application/src/main/kotlin/com/shareyourproxy/api/rx/command/eventcallback/UserContactAddedEventCallback.kt
package com.shareyourproxy.api.rx.command.eventcallback

import android.os.Parcel
import android.os.Parcelable

import com.shareyourproxy.api.domain.model.User

/**
 * User contact has been added.
 */
internal final class UserContactAddedEventCallback(user: User, val contactId: String) : UserEventCallback(user) {

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(contactId)
    }

    companion object {
        private val CL = UserContactAddedEventCallback::class.java.classLoader
        val CREATOR: Parcelable.Creator<UserContactAddedEventCallback> = object : Parcelable.Creator<UserContactAddedEventCallback> {
            override fun createFromParcel(parcel: Parcel): UserContactAddedEventCallback {
                return UserContactAddedEventCallback(
                        parcel.readValue(CL) as User, parcel.readValue(CL) as String)
            }

            override fun newArray(size: Int): Array<UserContactAddedEventCallback?> {
                return arrayOfNulls(size)
            }
        }
    }
}
